contracts/token/ERC7984/extensions/ERC7984Omnibus.sol

```diff
@@ -0,0 +1,200 @@
+ * balances externally.
+ */
+abstract contract ERC7984Omnibus is ERC7984 {
+    /**
+     * @dev Emitted when a confidential transfer is made representing the onchain settlement of
+     * an omnibus transfer from `sender` to `recipient` of amount `amount`. Settlement occurs between
+     * `omnibusFrom` and `omnibusTo` and is represented in a matching {IERC7984-ConfidentialTransfer} event.
+     *
+     * NOTE: `omnibusFrom` and `omnibusTo` get permanent ACL allowances for `sender` and `recipient`.
+     */
+    event OmnibusConfidentialTransfer(
+        address indexed omnibusFrom,
+        address indexed omnibusTo,
+        eaddress sender,
+        eaddress indexed recipient,
+        euint64 amount
+    );
+
+    /**
+     * @dev The caller `user` does not have access to the encrypted address `addr`.
+     *
+     * NOTE: Try using the equivalent transfer function with an input proof.
+     */
+    error ERC7984UnauthorizedUseOfEncryptedAddress(eaddress addr);
+
+    function confidentialTransferOmnibus(
+        address omnibusTo,
+        externalEaddress externalSender,
+        externalEaddress externalRecipient,
+        externalEuint64 externalAmount,
+        bytes calldata inputProof
+    ) public virtual returns (euint64) {
+        return
+            confidentialTransferFromOmnibus(
+                msg.sender,
+                omnibusTo,
+                externalSender,
+                externalRecipient,
+                externalAmount,
+                inputProof
+            );
+    }
+
+    function confidentialTransferOmnibus(
+        address omnibusTo,
+        eaddress sender,
+        eaddress recipient,
+        euint64 amount
+    ) public virtual returns (euint64) {
+        return confidentialTransferFromOmnibus(msg.sender, omnibusTo, sender, recipient, amount);
+    }
+
+    function confidentialTransferFromOmnibus(
+        address omnibusFrom,
+        address omnibusTo,
+        externalEaddress externalSender,
+        externalEaddress externalRecipient,
+        externalEuint64 externalAmount,
+        bytes calldata inputProof
+    ) public virtual returns (euint64) {
+        eaddress sender = FHE.fromExternal(externalSender, inputProof);
+        eaddress recipient = FHE.fromExternal(externalRecipient, inputProof);
+        euint64 amount = FHE.fromExternal(externalAmount, inputProof);
+
+        return _confidentialTransferFromOmnibus(omnibusFrom, omnibusTo, sender, recipient, amount);
+    }
+
+    function confidentialTransferFromOmnibus(
+        address omnibusFrom,
+        address omnibusTo,
+        eaddress sender,
+        eaddress recipient,
+        euint64 amount
+    ) public virtual returns (euint64) {
+        require(FHE.isAllowed(sender, msg.sender), ERC7984UnauthorizedUseOfEncryptedAddress(sender));
+        require(FHE.isAllowed(recipient, msg.sender), ERC7984UnauthorizedUseOfEncryptedAddress(recipient));
```

Comment:
No need to:
```suggestion
        require(FHE.isAllowed(recipient, msg.sender), ERC7984UnauthorizedUseOfEncryptedAddress(recipient));
        require(FHE.isAllowed(amount, msg.sender), ERC7984UnauthorizedUseOfEncryptedAmount(amount, msg.sender));
```
?
And update other event to `ERC7984UnauthorizedUseOfEncryptedAddress(euint64 addr, address user)`?